File: test/rs/case18_tuple_assign.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_floor, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

fn swap_sum_18(mut a: i64, mut b: i64) -> i64 {
    let mut x: i64 = a;
    let mut y: i64 = b;
    let __pytra_tuple_0_1 = y;
    let __pytra_tuple_1_2 = x;
    x = __pytra_tuple_0_1;
    y = __pytra_tuple_1_2;
    return ((x) + (y));
}

fn main() {
    py_print(swap_sum_18(10, 20));
}

File: test/cpp/case18_tuple_assign.cpp
// このファイルは `test/py/case18_tuple_assign.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case18_tuple_assign.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

int64 swap_sum_18(int64 a, int64 b) {
    int64 x = a;
    int64 y = b;
    py_swap(x, y);
    return x + y;
}

int main() {
    py_print(swap_sum_18(10, 20));
    return 0;
}
